### PR TITLE
Install test refactored

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This refactors the install tests to:
- Run on both Android and iOS (previously it was just on iOS)
- Run with the new RN architecture enabled and disabled
- Run with RN@latest and RN@next (in an attempt to catch bugs early)
- Run with both jsc and hermes enabled
- Run at 7AM CET every Monday (to help us prioritise any bug on the upcoming planning meeting)

### CODE REVIEW

File: .github/workflows/install-test-react-native.yml
Comment (kraenhansen) [inline .github/workflows/install-test-react-native.yml:137]: I will change this to "SLACK_TEAM_WEBHOOK" once it's merged.

**Comment by takameyer** [comment]
Could we send the results of this test as a report in Slack?  Otherwise, we will forget to check this regularly.

File: .github/workflows/install-test-react-native.yml
Comment (takameyer) [reply .github/workflows/install-test-react-native.yml:137]: What channel will this show up in?

File: .github/workflows/install-test-react-native.yml
Comment (takameyer) [inline .github/workflows/install-test-react-native.yml:80]: We could skip this for non-ios platforms

File: .github/workflows/install-test-react-native.yml
Comment (takameyer) [inline .github/workflows/install-test-react-native.yml:131]: Is this enough information?  Would be great to see a table of the matrix and what passed or failed.

File: install-tests/react-native/patches/ccache-Podfile.patch
Comment (takameyer) [inline install-tests/react-native/patches/ccache-Podfile.patch:10]: Where are we performing this swap?  Shouldn't there be a step somewhere to replace `clang` and `clang++` with a symbolic link to `ccache`?

File: install-tests/react-native/cli.ts
Comment (takameyer) [inline install-tests/react-native/cli.ts:163]: Maybe not an issue, but `npx pod-install` will automatically call `repo update` if this fails.

**Review by takameyer — APPROVED** [review]
Just a few clarifying questions and suggestions.  Nothing holding this back, except that it seems `realm-core` is being updated?

File: .github/workflows/install-test-react-native.yml
Comment (kraenhansen) [reply .github/workflows/install-test-react-native.yml:137]: `SLACK_TEST_WEBHOOK` goes to a private `#realm-js-bot-tests` channel and `SLACK_TEAM_WEBHOOK` goes to `#realm-js-team`.

File: .github/workflows/install-test-react-native.yml
Comment (kraenhansen) [reply .github/workflows/install-test-react-native.yml:131]: I think you're right .. I did try, but I can try a bit harder :)

File: install-tests/react-native/patches/ccache-Podfile.patch
Comment (kraenhansen) [reply install-tests/react-native/patches/ccache-Podfile.patch:10]: According to https://reactnative.dev/docs/build-speed#xcode-specific-setup this is enough to have XCode pick up the ccache executable if it's in the `PATH` and [that happens here](https://github.com/realm/realm-js/blob/kh/install-test-refactored/.github/workflows/install-test-react-native.yml#L62).

File: install-tests/react-native/cli.ts
Comment (kraenhansen) [reply install-tests/react-native/cli.ts:163]: Let's add this for good measures then.

```suggestion
        // Use --no-repo-update to avoid updating the repo if the install fails
        exec("pod", ["install", "--no-repo-update"], { cwd: path.resolve(appPath, "ios"), env });
```

File: install-tests/react-native/cli.ts
Comment (kneth) [inline install-tests/react-native/cli.ts:36]:
```suggestion
const TIMEOUT = 5 * 60 * 1000; // 5 min should be plenty of time from app has launched until message gets received
```

**Review by kneth — APPROVED** [review]

**Pull request merged**
